src/index.ts

```diff
@@ -183,19 +183,21 @@ const getTarEntryHandler = (
 		stream: NodeJS.ReadableStream,
 		next: (err?: Error) => void,
 	) => {
-		streamToPromise(stream).then((buffer: Buffer) => {
-			const name = normalizeTarEntry(header.name);
-			if (name === dockerfileName) {
-				const newDockerfile = transpose(buffer.toString(), opts);
-				pack.entry({ name: 'Dockerfile' }, newDockerfile);
-			} else {
-				if (name === opts.hostQemuPath && opts.qemuFileMode) {
-					header.mode = opts.qemuFileMode;
+		streamToPromise(stream)
+			.then((buffer: Buffer) => {
+				const name = normalizeTarEntry(header.name);
+				if (name === dockerfileName) {
+					const newDockerfile = transpose(buffer.toString(), opts);
+					pack.entry({ name: dockerfileName }, newDockerfile);
```

Comment:
@CameronDiver, @dfunckt, suppose the user specifies an alternative Dockerfile, say "Dockerfile.debug" instead of the usual "Dockerfile".  This module was already correctly selecting "Dockerfile.debug" for transposing, but the result of transposing was always being saved to a tar entry named "Dockerfile". Can you think of any reason why the specified Dockerfile name should not be preserved in the output stream? I've considered the possibility that some old Docker daemons might ignore a specified Dockerfile name and always look for "Dockerfile" instead, but it looks like Docker API versions [at least as old as 1.18 (currently in 1.40)](https://docs.docker.com/engine/api/version-history/) already supported specified Dockerfile names.

I am sure that this change is part of the fix of an actual reported problem (balena-io/balena-cli/issues/1624), but I am less sure whether it could also introduce some side effect. I've also considered an alternative solution: the CLI could refrain from passing on the specified Dockerfile name to the Docker daemon, so it's always "Dockerfile". But then we'd also have to handle the case where the input tar stream contained both a "Dockerfile" and a "Dockerfile.debug" file, in which case the output tar stream might end up with two entries for "Dockerfile" -- we would have to add code to exclude in the input Dockerfile from the output. What this PR currently has seems to be the cleaner fix, if side effects were not expected.

If the diff is not clear:
**Before:** `pack.entry({ name: 'Dockerfile' }, newDockerfile);`
**After:** `pack.entry({ name: dockerfileName }, newDockerfile);`

Reply:
Sounds sensible to me. As long as clients remember to send the Dockerfile name explicitly to the daemon this shouldn't cause problems. @CameronDiver has a better idea of the build pipeline than me however and should be able to confirm.

Reply:
> As long as clients remember to send the Dockerfile name explicitly to the daemon

Thanks - yes, the CLI does; which other clients should we consider? I gather that `balena-builder` does not use `docker-qemu-transpose` (not even as a transitive dependency), and in any case it sets the specified Dockerfile name through the multibuild module:
https://github.com/balena-io-modules/resin-multibuild/blob/v4.4.1/lib/build.ts#L140
Any client that uses `resin-multibuild` would be able to send the specified dockerfile name to the daemon.

Reply:
For what it is worth, I have tested this PR with CLI PR balena-io/balena-cli/pull/1628 and it worked fine. Together they fix CLI issue balena-io/balena-cli/issues/1624.

Reply:
@pdcastro I think the main thing to remember is that this module massively predates the ability to specify a Dockerfile when building with any of our build pipelines, which is why it always wrote out to `Dockerfile`. I think this PR makes a lot of sense :+1: